packages/jsxstyle-utils/src/getStyleKeysForProps.ts

```diff
@@ -52,7 +52,14 @@ export default function getStyleKeysForProps(props, pretty = false) {
   let usesMediaQueries = false;
 
   // return value
-  const styleKeyObj = {};
+  const styleKeyObj: {
```

Comment:
You should extract this type and explicitly use it as the return value on the function.  That will make the function signature easier to skim, and keep you from having to cast an empty dictionary to match a type it doesn't yet implement.

Reply:
What type would I assign to `styleKeyObj` in that case? I’m not sure how to initialise an empty object with a type in TypeScript without making all the keys optional (which seems wrong to me? idk tho).

Reply:
I'm not sure if you would need to set it to anything.  I think the type could be inferred in the function body and explicitly set at the function boundary.

```typescript
export type StyleKey = {
  styles: string,
  mediaQuery?: string,
  pseudoclass?: string,
  pseudoelement?: string,
};

export default function getStyleKeysForProps(props: any, pretty = false): Dict<StyleKey> {
  const result = {};
  // build result
  return result
}
```

Reply:
I was getting this error when I tried that earlier:
```
Type '{}' is not assignable to type 'StyleKeyObject | null'.
```

Reply:
Maybe I was too optimistic about what it could infer.  I was sleepy and on a plane when I wrote that (and I'm still sleepy and just-off a plane), so I hadn't actually tested it.

In that case, probably:

```typescript
export type StyleKey = {
  styles: string,
  mediaQuery?: string,
  pseudoclass?: string,
  pseudoelement?: string,
};

export default function getStyleKeysForProps(props: any, pretty = false): Dict<StyleKey> {
  const result: Dict<StyleKey> = {};
  // build result
  return result
}
```
You don't even have to `Partial<>` it, because a `Dict` can have no members and still be a `Dict`.

On a related note, I think you're always supposed to give explicit types to the arguments and return value of any exported function.  I know that's true in Flow, but I only presume so in TS.  It's supposed to simplify the amount of work the typechecker has to do if you annotate at the boundaries, which makes type-checking easier to parallelize.

Reply:
Heh, no problem. Yeah, that makes sense to use an explicit return type. I wasn’t wild about the inline multiline type definition anyway. Much nicer to have it top-level.